### PR TITLE
ENH: effect_group (ATET/ATC) for TreatmentEffect.aipw and aipw_wls

### LINKED ISSUE #10226: ENH: effect_group (ATET/ATC) for TreatmentEffect.aipw and aipw_wls
### Is your feature request related to a problem? Please describe

TreatmentEffect.ra, .ipw and .ipw_ra take effect_group ("all"/treated/control), but .aipw and .aipw_wls don't they only return ATE. The aipw_wls docstring says "Option for effect on treated or on untreated is not available" and test_teffects.py skips the ATET block with # not available for aipw. Stata added atet to teffects aipw in version 18, which is presumably why it was left out originally.

### Describe the solution you'd like

Add `effect_group` to `aipw` and `aipw_wls` with the same accepted values as the other estimators (`"all"`, `1`/`"treated"`, `0`/`"untreated"`/`"control"`).

ATET needs its own effect estimating equations rather than a restriction of the ATE moments to the treated. For the treated, the counterfactual control mean combines the outcome-model prediction averaged over treated observations with an augmentation term from the control residuals weighted by the propensity odds $p/(1-p)$:

$$
\hat\mu_{0|T=1} = \frac{1}{n_1}\sum_i \Big[ T_i\,\hat\mu_0(x_i) + (1-T_i)\,\frac{\hat p_i}{1-\hat p_i}\,\{y_i-\hat\mu_0(x_i)\} \Big],
\qquad
\hat\mu_{1|T=1} = \frac{1}{n_1}\sum_i T_i\,y_i
$$

ATC is symmetric. The GMM effect moments keep both components, stacked with the outcome and treatment model scores as now, so the standard errors account for the first-step estimation. This is the standard doubly robust ATT estimating equation (Hahn 1998; Moodie, Saarela & Stephens 2018) and corresponds to what Stata 18 added to `teffects aipw, atet`.

For `aipw_wls` the WNLS outcome-model weights s

### Describe alternatives you have considered

_No response_

### References

- Hahn, J. (1998). "On the Role of the Propensity Score in Efficient Semiparametric Estimation of Average Treatment Effects." Econometrica 66(2), 315–331. doi:10.2307/2998560 
- Moodie, E.E.M., Saarela, O. & Stephens, D.A. (2018). "A doubly robust weighting estimator of the average treatment effect on the treated." Stat 7(1), e205. doi:10.1002/sta4.205
- StataCorp (2023), Stata 18 [CAUSAL] teffects aipw, “Methods and formulas”

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kirilklein**
Adds `effect_group` to `TreatmentEffect.aipw` and `aipw_wls`, supporting effects on the treated (ATET) and controls (ATC). The default remains the overall ATE. Both point estimates and joint GMM inference support the new targets. Existing positional calls retain their meaning; `effect_group` is keyword-only.

Stata 18 added `atet` to `teffects aipw`, but I don't have access to it. I verified the point estimates against DoubleML 0.11.4 instead, using `DoubleMLIRM(score="ATTE")`, which solves the same ATT estimating equation.

The comparison uses full-sample fitting without cross-fitting, the same unpenalized logit propensity specification, and propensity scores clipped to `[0.01, 0.99]`. ATC is obtained by reversing treatment labels and negating the estimate.

| Estimator | Target | DoubleML | statsmodels (this PR) | rel. diff |
|---|---|---:|---:|---:|
| `aipw` | ATET | -226.178244783 | -226.178244812 | 1.3e-10 |
| `aipw` | ATC | -230.764613267 | -230.764613251 | 7.1e-11 |
| `aipw_wls` | ATET | -223.448320722 | -223.448320722 | < 1e-12 |
| `aipw_wls` | ATC | -223.602248802 | -223.602248802 | < 1e-12 |

For `aipw`, DoubleML fits its own propensity and OLS outcome models. For `aipw_wls`, DoubleML evaluates its score using the statsmodels WLS predictions, so that comparison validates the score calculation rather than independently validating the WLS fit.

The reference tests cover point estimates only. Standard errors use the existing statsmodels joint GMM approach and are not benchmarked against DoubleML.

The reference values are stored in `results_teffects.py` and tested in `test_aipw_effect_group_doubleml`. The comparison can be reproduced with DoubleML 0.11.4 and scikit-learn 1.9.0 by running:

```bash
python -m statsmodels.treatment.tests.results.teffects_doubleml
```

The script specifies the solver, tolerance, sample splitting, and weight normalization settings. If a reviewer can produce Stata 18 output, I'm happy to add it as an additional independent reference.

- [x] closes #10226
- [x] tests added / passed.
- [x] code/documentation is well formatted.
- [x] properly formatted commit message.

#### AI Disclosure

- [x] AI tools were used.
      Tool(s): Claude, Codex.
      Used for: Claude drafted the implementation and tests. Codex reviewed the mathematics and API compatibility, fixed positional-argument compatibility, added regression tests and a reference-generation script, and edited the PR description.
      I have personally read, understood, and can explain every line of this diff, and
      have verified the statistical/numerical correctness of the change.

### CODE REVIEW

File: statsmodels/treatment/tests/results/teffects_doubleml.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from sklearn.preprocessing import StandardScaler
+
+from statsmodels.discrete.discrete_model import Logit
+from statsmodels.regression.linear_model import OLS
+from statsmodels.treatment.treatment_effects import TreatmentEffect
+
+from .results_teffects import results_aipw_atet_dml
+
+
+def main():
```
Comment (bashtage) [inline statsmodels/treatment/tests/results/teffects_doubleml.py]: Please convert this from a script requiring `main()` after a `__name__ == "__main__"` to just be a straight script that executes all code when run.  Using `main(),` while often seen as a good practice in general, is not how we write single-purpose scripts.
